torch/nn/functional.py

```diff
@@ -5206,8 +5206,8 @@ def multi_head_attention_forward(
     attn_mask = _canonical_mask(
         mask=attn_mask,
         mask_name="attn_mask",
-        other_type=_none_or_dtype(key_padding_mask),
-        other_name="key_padding_mask",
```

Comment:
Is there a reason to change `other_name`? (I guess it would affect serialization, wouldn't it?)

Reply:
See description.  This is for cross checking types of masks.  After mask1 and mask2 have been cross checked, and mask1 has been converted, we would fail the check if we repeat the check at that point.